tests/ctl/core/test_api.py

```diff
@@ -1151,6 +1453,88 @@ def test_system_update_privacy_declaration_invalid_duplicate(
             and system.privacy_declarations[1].name == "new declaration 1"
         )
 
+    def test_system_update_dictionary_fields(
+        self,
+        test_config,
+        system_update_request_body_with_new_dictionary_fields,
+        system,
+        db,
+        generate_system_manager_header,
+    ):
+        assert system.name != self.updated_system_name
+
+        auth_header = generate_system_manager_header([system.id])
+        result = _api.update(
+            url=test_config.cli.server_url,
+            headers=auth_header,
+            resource_type="system",
+            json_resource=system_update_request_body_with_new_dictionary_fields.json(
+                exclude_none=True
+            ),
+        )
+
+        assert result.status_code == HTTP_200_OK
+
+        db.refresh(system)
+
+        assert system.name == self.updated_system_name
+        assert system.vendor_id == "test_vendor"
```

Comment:
Not asking for a change here, just giving my thoughts

When the asserts are hardcoded like so, it can make it very hard to add additional test cases down the road using something like parametrization

If we figure out there is an edge case, instead of adding another system that catches it and letting the test do its thing, we now have to copy/pasta the entire test

Keeping tests DRY isn't a priority, verbosity is helpful, but in this case I think it decreases future testability